api/src/test/java/org/openmrs/module/fhir2/providers/r3/ConditionFhirR3ResourceProviderTest.java

```diff
@@ -139,14 +146,53 @@ public void getConditionHistoryByWithWrongId_shouldThrowResourceNotFoundExceptio
 	
 	@Test
 	public void createCondition_shouldCreateNewCondition() {
-		when(conditionService.saveCondition(any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(condition);
-		
+
+		when(conditionService.create(any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(condition);
+
+// 		when(conditionService.saveCondition(any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(condition);
+
 		MethodOutcome result = resourceProvider.createCondition(Condition30_40.convertCondition(condition));
 		
 		assertThat(result, notNullValue());
 		assertThat(result.getCreated(), is(true));
+
+		assertThat(result.getResource(), CoreMatchers.equalTo(condition));
+	}
+	
+	@Test(expected = InvalidRequestException.class)
+	public void updateCondition_shouldThrowInvalidRequestForUuidMismatch() {
+		when(conditionService.update(eq(WRONG_CONDITION_UUID), any(org.hl7.fhir.r4.model.Condition.class)))
+		        .thenThrow(InvalidRequestException.class);
+		
+		resourceProvider.updateCondition(new IdType().setValue(WRONG_CONDITION_UUID),
+		    Condition30_40.convertCondition(condition));
+	}
+	
+	@Test(expected = InvalidRequestException.class)
+	public void updateCondition_shouldThrowInvalidRequestForMissingId() {
+		org.hl7.fhir.r4.model.Condition noIdCondition = new org.hl7.fhir.r4.model.Condition();
+		
+		when(conditionService.update(eq(CONDITION_UUID), any(org.hl7.fhir.r4.model.Condition.class)))
+		        .thenThrow(InvalidRequestException.class);
+		
+		resourceProvider.updateCondition(new IdType().setValue(CONDITION_UUID),
+		    Condition30_40.convertCondition(noIdCondition));
+	}
+	
+	@Test(expected = MethodNotAllowedException.class)
+	public void updateCondition_shouldThrowMethodNotAllowedIfDoesNotExist() {
+		
+		org.hl7.fhir.r4.model.Condition wrongCondition = new org.hl7.fhir.r4.model.Condition();
+		
+		wrongCondition.setId(WRONG_CONDITION_UUID);
+		when(conditionService.update(eq(WRONG_CONDITION_UUID), any(org.hl7.fhir.r4.model.Condition.class)))
+		        .thenThrow(MethodNotAllowedException.class);
+		
+		resourceProvider.updateCondition(new IdType().setValue(WRONG_CONDITION_UUID),
+		    Condition30_40.convertCondition(wrongCondition));
 		assertThat(result.getResource(), notNullValue());
 		assertThat(result.getResource().getIdElement().getIdPart(), equalTo(CONDITION_UUID));
+
```

Comment:
These lines seem to be causing the failure, but we don't need them as we should expect the `updateCondition()` call to throw the error we expect to catch in this method. So it should be safe to delete them.

```suggestion
```